docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -338,19 +338,182 @@ The result will look like this:
+
+The default context is resolved depending on various factors:
+- If an absolute path is specified:
+   - in a Git repository (defined by the presence of a `.git` sub-directory), the default context is the root of the Git repository.
+   - in a non-repository location (defined by the absence of a `.git` sub-directory), the default context is the directory containing a `dagger.json` file.
+- If a relative path is specified
+  - the default context is always the directory containing a `dagger.json` file.
+
+Here is an example of a Dagger Function with a default context for its directory argument:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+</TabItem>
+<TabItem value="Python">
+</TabItem>
+<TabItem value="TypeScript">
+```typescript file=./snippets/functions/arguments-contextual-dir/typescript/index.ts
+```
+</TabItem>
+</Tabs>
+
+Assuming the directory structure below, the following table describes how the argument is resolved:
+
+```shell
+.
+├── README.md
+├── my-module
+│   ├── sdk
+│   ├── ...
+├── dagger.json
+├── .git
+│   └── ...
+├── index.html
+├── public
+│   └── ...
+├── src
+│   └── ...
+```
+
+| Location | Default context ('defaultPath`) | Dagger Function invocation | `source` resolves to |
+| --- | --- | --- | --- |
+| Git repository | `/` | `dagger call read-dir` | `/` |
+| Git repository | `/src` | `dagger call read-dir` | `/src` |
+| Git repository | `.` | `dagger call read-dir` | `/my-module` |
+| Git repository | `..` | `dagger call read-dir` | `/` |
+| Non-Git directory | `/` | `dagger call read-dir` | `/` |
+| Non-Git directory | `/src` | `dagger call read-dir` | `/src` |
+| Non-Git directory | `.` | `dagger call read-dir` | `/my-module` |
+| Non-Git directory | `..` | `dagger call read-dir` | `/` |
+
+If an argument is explicitly passed to the Dagger Function, it always overrides the default context.
+
+| Default context ('defaultPath`) | Dagger Function invocation | `source` resolves to |
+| --- | --- | --- |
+| `/` | `dagger call read-dir --source=public` | `/public` |
+| `/src` | `dagger call read-dir --source=public` | `/public` |
+| `.` | `dagger call read-dir --source=public` | `/public` |
+| `..` | `dagger call read-dir --source=public` | `/public` |
+
+
+:::tip
+It's also possible to provide an `ignore` parameter to a contextual argument of type `Directory` to automatically ignore or include files in the directory. The `ignore` parameter follows the `.gitignore` syntax: prefixing a path with `!` will actually include it instead of ignoring it.
+:::
+
+
+/* TBC
+The context is set through the `defaultPath` parameter metadata pointing to a location inside the module's context directory.
+
+- If the path is absolute, its root is the path to the git repository containting that module (called `context directory`), falling into the location of the `dagger.json` if no `.git` directory is found.
+- If the path is relative, its root is the path to `dagger.json` location (called `root directory`).
```

Comment:
Can we check this statement? If my dagger.json is in the Git repo root and my module source is in /my-module, then using a relative path like `.` returns the module source dir and not the repo root (where dagger.json is).

Reply:
Works as expected with 0.13

```
.
├── LICENSE
├── README.md
├── cypress.config.ts
├── dagger.json
├── env.d.ts
├── index.html
├── my-module
├── package-lock.json
├── package.json
├── public
├── src
├── tsconfig.app.json
├── tsconfig.node.json
├── tsconfig.vitest.json
├── vite.config.ts
└── vitest.config.ts

$ dagger version
dagger v0.13.0 (registry.dagger.io/engine:v0.13.0) linux/amd64

  @func()
  async readDir(
    @argument({ defaultPath: "." }) source: Directory,
  ): Promise<string[]> {
    return await source.entries()
  }

$ dagger call read-dir
.aa
.eslintrc.cjs
.gitattributes
.gitignore
.prettierrc.json
.vscode
LICENSE
README.md
cypress.config.ts
dagger.json
env.d.ts
index.html
my-module
package-lock.json
package.json
public
src
tsconfig.app.json
tsconfig.node.json
tsconfig.vitest.json
vite.config.ts
vitest.config.ts
```